src/song.c

```diff
@@ -531,19 +531,19 @@ mpd_song_feed(struct mpd_song *song, const struct mpd_pair *pair)
 	}
 
 	if (strcmp(pair->name, "Time") == 0)
-		mpd_song_set_duration(song, atoi(pair->value));
+		mpd_song_set_duration(song, strtoul(pair->value, NULL, 10));
 	else if (strcmp(pair->name, "duration") == 0)
-		mpd_song_set_duration_ms(song, 1000 * atof(pair->value));
+		mpd_song_set_duration_ms(song, 1000 * strtoul(pair->value, NULL, 10));
```

Comment:
This truncates and discards the fractional part of the song duration.
Again, you truncate too early!